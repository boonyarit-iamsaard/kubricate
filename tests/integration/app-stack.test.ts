import path from 'node:path';
import { afterEach, describe, expect, it } from 'vitest';
import { rimraf } from 'rimraf';
import { executeKubricate } from '../helpers/execute-kubricate';
import fs from 'node:fs/promises';

const rootDir = path.resolve(__dirname, '..');
const fixturesDir = path.join(rootDir, 'fixtures', 'generate-output-stack');
const outputDir = path.join(fixturesDir, 'output');

describe('CLI Integration (generate)', () => {
  
  it('should generate expected files', async () => {
    const args = [
      'generate',
      '--root',
      fixturesDir,
    ];

    const { stdout, exitCode } = await executeKubricate(args, { reject: false });

    expect(exitCode).toBe(0);
    expect(stdout).toContain('Generating stacks');

    const files = await fs.readdir(outputDir);

    for (const file of files) {
      const fullPath = path.join(outputDir, file);
      const content = await fs.readFile(fullPath, 'utf-8');

      // âœ… Snapshot EACH file individually
      expect(content).toMatchSnapshot(file);
    }
  });
});