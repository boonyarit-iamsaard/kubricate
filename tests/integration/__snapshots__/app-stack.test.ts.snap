// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CLI Integration (generate) > should generate expected files > namespace.yml 1`] = `
"apiVersion: v1
kind: Namespace
metadata:
  name: my-namespace
  labels:
    thaitype.dev/kubricate: "true"
    thaitype.dev/kubricate/stack-id: namespace
    thaitype.dev/kubricate/resource-id: namespace
  annotations:
    thaitype.dev/kubricate/stack-name: NamespaceStack
    thaitype.dev/kubricate/resource-hash: 446fa07a363d589aef6d97e7338f6d74f9fd397d9cec5209be5fd4b2b26bd3e0
---
"
`;
